iceoryx2-bb/log/src/lib.rs

```diff
@@ -185,6 +188,45 @@ pub enum LogLevel {
     Fatal = 5,
 }
 
+impl LogLevel {
+    fn from_str_fuzzy(s: &str) -> LogLevel {
+        match s.to_lowercase().as_str() {
+            "trace" => LogLevel::Trace,
+            "debug" => LogLevel::Debug,
+            "info" => LogLevel::Info,
+            "warn" => LogLevel::Warn,
+            "error" => LogLevel::Error,
+            "fatal" => LogLevel::Fatal,
+            _ => {
+                println!("Error: unknown logging level {:?}", s);
+                println!("Warning: setting log level as : {:?}", DEFAULT_LOG_LEVEL);
+                DEFAULT_LOG_LEVEL
+            }
+        }
+    }
+}
+
+/// Sets the log level by reading environment variable "IOX2_LOG_LEVEL" or default it wiht LogLevel::INFO
+pub fn set_log_level_from_env_or_default() {
+    let log_level = env::var("IOX2_LOG_LEVEL")
+        .ok()
+        .map(|s| LogLevel::from_str_fuzzy(&s))
+        .unwrap_or(DEFAULT_LOG_LEVEL);
+    ENV_LOG_LEVEL.set(log_level).unwrap();
+    set_log_level(*ENV_LOG_LEVEL.get().unwrap());
+}
+
+/// Sets the log level by reading environment variable "IOX2_LOG_LEVEL", and if the environment variable
+/// doesn't exits it sets it with a user-defined logging level
+pub fn set_log_level_from_env_or(v: LogLevel) {
+    let log_level = env::var("IOX2_LOG_LEVEL")
+        .ok()
+        .map(|s| LogLevel::from_str_fuzzy(&s))
+        .unwrap_or(v);
+    ENV_LOG_LEVEL.set(log_level).unwrap();
+    set_log_level(*ENV_LOG_LEVEL.get().unwrap());
```

Comment:
With the latest changes, I think you can remove `ENV_LOG_LEVEL` entirely and just use
```suggestion
    set_log_level(log_level);
```